addons/purchase_stock/models/stock_rule.py

```diff
@@ -146,6 +146,19 @@ def _run_buy(self, procurements):
                         procurement.values, po))
             self.env['purchase.order.line'].sudo().create(po_line_values)
 
+            order_point = po.order_line.mapped('orderpoint_id')
+            if order_point:
+                odoobot = self.env.ref('base.partner_root')
+                origin = order_point[0].display_name
+                po_origin = list(po.origin.split(', ')) if po.origin else []
+
+                if origin not in po_origin:
+                    po.write({'origin': po.origin + ', ' + ', '.join([origin]) if po.origin else ', '.join([origin])})
```

Comment:
I don't understand why call `join` on a list containing only one string ? That seems useless, no ?
It should be that I think:
```suggestion
                    po.write({'origin': ', '.join([po.origin, origin]) if po.origin else origin})
```
Or even better:
```suggestion
                    po.write({'origin': f'{po.origin}, {origin}' if po.origin else origin})
```